napari/utils/events/evented_model.py

```diff
@@ -234,19 +268,33 @@ def __setattr__(self, name: str, value: Any) -> None:
 
         # grab current value
         before = getattr(self, name, object())
+        before_deps = {}
+        with warnings.catch_warnings():
+            # we still need to check for deprecated properties
+            warnings.simplefilter("ignore", DeprecationWarning)
```

Comment:
As catch_warnings is not thread-safe, there is a need to work around it. But as it will not be simple, then there is a need to open an Issue describing the problem and solve it in a follow-up PR. 

My idea is to define `deprecated_property` that will use instance-specific variable to control if emit deprecation/future warning.